google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/GapicSpannerRpc.java

```diff
@@ -140,25 +147,55 @@ public GapicSpannerRpc(SpannerOptions options) throws IOException {
                 .build());
     CredentialsProvider credentialsProvider =
         GrpcTransportOptions.setUpCredentialsProvider(options);
-
-    this.stub =
+    try {
+      // TODO: bump the version of gax and remove this try-catch block
+      // applyToAllUnaryMethods does not throw exception in the latest version
+      this.stub =
         GrpcSpannerStub.create(
             SpannerStubSettings.newBuilder()
                 .setTransportChannelProvider(channelProvider)
                 .setCredentialsProvider(credentialsProvider)
+                .applyToAllUnaryMethods(
+                    new ApiFunction<UnaryCallSettings.Builder<?, ?>, Void>() {
+                      @Override
+                      public Void apply(UnaryCallSettings.Builder<?,?> builder) {
```

Comment:
can you add a comment here explaining why we are turning off retries.

Reply:
Sure.